homeassistant/components/sensor/mfi.py

```diff
@@ -0,0 +1,104 @@
+"""
+homeassistant.components.sensor.mfi
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Ubiquiti mFi Sensors.
+
+Configuration:
+
+sensor:
+  platform: mfi
+  host: ADDRESS
+  port: PORT
+  username: USERNAME
+  password: PASSWORD
+
+Variable:
+
+host
+*Required
+ADDRESS is the IP or hostname of your mFi controller.
+
+port
+*Optional
+PORT is the port of your mFi controller (usually 6443)
+
+username
+*Required
+USERNAME is the mFi admin username
+
+password
+*Required
+PASSWORD is the mFi admin user's password
+"""
+
+import logging
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['mficlient==0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_MODELS = [
+    'Ubiquiti mFi-THS',
+    'Ubiquiti mFi-CS',
+    'Outlet',
+]
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up mFi sensors. """
+
+    host = config.get('host')
+    port = int(config.get('port', 6443))
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
```

Comment:
Can you please check if the mandatory configuration parts are provided? If not, the setup should fail with a log entry.